src/aws/flb_aws_credentials_sts.c

```diff
@@ -293,8 +295,12 @@ struct flb_aws_provider *flb_sts_provider_create(struct flb_config *config,
         goto error;
     }
 
-    implementation->endpoint = flb_aws_endpoint("sts", region);
-    if (!implementation->endpoint) {
+    if (sts_endpoint) {
+        implementation->custom_endpoint = FLB_TRUE;
+        implementation->endpoint = removeProtocol(sts_endpoint, "https://");
+    }
+    else {
+        implementation->custom_endpoint = FLB_FALSE; 
         goto error;
     }
```

Comment:
Am I reading this code wrong... because it looks like if there is no custom endpoint, then we `goto error` and fail initialization...